SAW compilation failure with GHC 9.2.7

On the master branch, SAW fails to compile using GHC 9.2.7.  The only error I can find is:
```
/Users/ramsdell/rep/saw-script/dist-newstyle/build/x86_64-osx/ghc-9.2.7/saw-core-0.1/build/Verifier/SAW/Lexer.hs:503:19: error: [-Werror]
    accepting non-standard pattern guards (use PatternGuards to suppress this message)
        (tagToEnum# (offset >=# 0#)),
        check <- alexIndexInt16OffAddr alex_check offset,
        (tagToEnum# (check ==# ord_c))
    |
503 |                   | GTE(offset,ILIT(0))
    |                   ^^^^^^^^^^^^^^^^^^^^^...
```


[all.txt](https://github.com/GaloisInc/saw-script/files/11222959/all.txt)

--- a/saw-core/src/Verifier/SAW/Lexer.x
+++ b/saw-core/src/Verifier/SAW/Lexer.x
@@ -2,6 +2,7 @@
 {-# LANGUAGE BangPatterns #-}
 {-# LANGUAGE DeriveFunctor #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE PatternGuards #-}
 {-# LANGUAGE ViewPatterns #-}
 
 {-# OPTIONS_GHC -fno-warn-missing-signatures #-}

